pkg/agent/multicast/mcast_discovery.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"antrea.io/libOpenflow/openflow13"
+	"antrea.io/libOpenflow/protocol"
+	"antrea.io/libOpenflow/util"
+	"antrea.io/ofnet/ofctrl"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+)
+
+const (
+	IGMPProtocolNumber = 2
+)
+
+const (
+	// queryInterval is the interval to send IGMP query messages.
+	queryInterval = time.Second * 125
+	// mcastGroupTimeout is used to identify the multicast group in which no local members have joined.
```

Comment:
"is the timeout to detect a group as stale if no IGMP report is received within the time"?

Reply:
updated.